Week-6/6.6.js

```diff
@@ -0,0 +1,29 @@
+function threeNumberSum(array, targetSum){
+    let result = [];
+    array.sort((a,b) => a-b);
+    for(let i = 0; i < array.length; i++){
+        let left = i + 1;
+        let right = array.length - 1;
+        while(left < right){
+            let currentSum = array[i] + array[left] + array[right];
+            if(currentSum === targetSum){
+                result.push([array[i], array[left], array[right]]);
+                left++;
+                right--;
+            } else if(currentSum < targetSum){
+                left++;
+            } else if(currentSum > targetSum){
+                right--;
+            }
+        }
+    }
+    return result;
+
+}
+
+// Time Complexity: O(n^2)
```

Comment:
try to reduce the time complexity.

Reply:
I checked It would be on a minimum of 2 loops. so this approach is fine.

Reply:
yeah I wasnt't finding any solution